### PR TITLE
[Doc] Deciding test nature markdown added

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
### All Submissions:

Description: The added markdown helps people to understand what points are to be kept in mind for a test case to be marked as disruptive/non-disruptive.

Fixes: #924

### CODE REVIEW

File: docs/BP/TestNature.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+2. Non-Disruptive test cases (hereon referred to as NDtc)
+
+Based on the design of redant, NDtcs are run paralelly while Dtcs are run sequentially. So, how does one decide whether the test case they're writing is a NDtc or a Dtc ?
+
+Following are some pointers which describe a Dtc. If a test case satisfies even one of the stated points, it is Dtc. For a test case to be NDtc, it shouldn't satisfy any of the stated points.
+
+* The test case reboots a node ( server or client ).
+* The test case reconfigures ports or changes firewalld rules.
+* The test case is unmounting the mountpoint which acts as glusterfs brick roots.
+* The test case aims to create a huge amount of volumes, snapshots, geo-rep sessions or gfind sessions.
```
Comment (nik-redhat) [inline docs/BP/TestNature.md:14]: Also:
- The test case performs any cluster-level operations

Comment (nik-redhat) [inline docs/BP/TestNature.md:14]: And, consumes high memory or performs huge IO (can be written in a better way :p)

File: docs/BP/TestNature.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+1. Disruptive test cases (hereon referred to as Dtc)
+2. Non-Disruptive test cases (hereon referred to as NDtc)
+
+Based on the design of redant, NDtcs are run paralelly while Dtcs are run sequentially. So, how does one decide whether the test case they're writing is a NDtc or a Dtc ?
+
+Following are some pointers which describe a Dtc. If a test case satisfies even one of the stated points, it is Dtc. For a test case to be NDtc, it shouldn't satisfy any of the stated points.
+
+* The test case reboots a node ( server or client ).
+* The test case reconfigures ports or changes firewalld rules.
+* The test case is unmounting the mountpoint which acts as glusterfs brick roots.
```
Comment (nik-redhat) [inline docs/BP/TestNature.md:13]: mountpoint or the volume bricks itself.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: docs/BP/TestNature.md (hunk 1 above)
Comment (srijan-sivakumar) [reply docs/BP/TestNature.md:14]: Ack

File: docs/BP/TestNature.md (hunk 2 above)
Comment (srijan-sivakumar) [reply docs/BP/TestNature.md:13]: Ack

File: docs/BP/TestNature.md (hunk 1 above)
Comment (srijan-sivakumar) [reply docs/BP/TestNature.md:14]: Hmm..this is a little vague. Even if we state consumes high memory, the question would be what is the value we consider as high memory and huge IO. If someone with say an env of 3 servers with 2 GB RAM, a disk per server for brick and 2 CPU cores were to run then the definition will change. I guess for now we can prop up volume count and other counts as a parameter for high memory consumption, till we actually benchmark and find a value which is the actual limitation..

Comment (srijan-sivakumar) [reply docs/BP/TestNature.md:14]: So, maybe we can just provide the example of our env along with the volume counts till which we consider it non disruptive. Any pointers on till what count have you tried out non disruptive tests ?

Comment (nik-redhat) [reply docs/BP/TestNature.md:14]: Make sense :thinking: ...... what I have seen is that we can get a maximum of 75 distribute (x3) volumes in a 3 server env having 2GB RAM and 2 CPUs. But again, adding such data won't make much sense either, so I think guess the current point can be kept as it is for the time being.

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**